types/node/events.d.ts

```diff
@@ -451,10 +455,52 @@ declare module "events" {
              */
             signal?: AbortSignal | undefined;
         }
+
+        export interface EventEmitterReferencingAsyncResource extends AsyncResource {
+            readonly eventEmitter: EventEmitterAsyncResource;
+        }
+
+        export interface EventEmitterAsyncResourceOptions extends AsyncResourceOptions, EventEmitterOptions {
+            /**
+             * The type of async event, this is required when instantiating `EventEmitterAsyncResource`
+             * directly rather than as a child class.
+             * @default new.target.name
+             */
+            name?: string;
+        }
+
+        /**
+         * Integrates `EventEmitter` with `AsyncResource` for `EventEmitter`s that require
+         * manual async tracking. Specifically, all events emitted by instances of
+         * `EventEmitterAsyncResource` will run within its async context.
+         *
+         * The EventEmitterAsyncResource class has the same methods and takes the
+         * same options as EventEmitter and AsyncResource themselves.
+         * @throws if `options.name` is not provided when instantiated directly.
+         * @since v17.4.0, v16.14.0
+         */
+        export class EventEmitterAsyncResource extends EventEmitter {
+            constructor(options: EventEmitterAsyncResourceOptions);
```

Comment:
```suggestion
            constructor(options?: EventEmitterAsyncResourceOptions);
```

I think `options` are optional and a fallback to default values will work in a lot of use cases.

Reply:
I do not have clear stance on this (quote about `name?`)

> The type of async event, this is required when instantiating `EventEmitterAsyncResource`

```console
Uncaught:
TypeError [ERR_INVALID_ARG_TYPE]: The "options.name" property must be of type string. Received undefined
```

Reply:
Then the default value for `name` is not `new.target.name`. And `name` is not optional.

Reply:
```js
import events from 'events'

class MyEmitter extends events.EventEmitterAsyncResource {}

const emitter = new MyEmitter()
```
work.

And
```
import events from 'events'

const emitter = new events.EventEmitterAsyncResource()
```
don't

So the comments in the PR are accurate.

Reply:
This is accurate, will fix.